src/main/java/org/example/service/ApartmentService.java

```diff
@@ -30,8 +30,9 @@ public List<Apartment> findAllApartmentsByRealtyAgent(int id) {
         return apartmentRepository.findAllByRealtyAgent_Id(id);
     }
 
-    public Apartment saveApartment(ApartmentDTO apartmentDTO) {
+    public Apartment saveApartment(int buildingId, ApartmentDTO apartmentDTO) {
         Apartment apartment = new Apartment();
+        apartment.setBuilding(buildingRepository.findById(buildingId).orElse(null));
```

Comment:
It's not correcr to use orElse(null) here, since we can get saving error
It's better to use .get() instead

PLEASE, check it in other services too